src/Common/REST/TEC/V1/Parameter_Types/Hex_Color.php

```diff
@@ -31,7 +32,31 @@ public function get_format(): ?string {
 	 * @inheritDoc
 	 */
 	public function get_validator(): ?Closure {
-		return $this->validator ?? fn( $value ): bool => is_string( $value ) && ( preg_match( '/^#([0-9a-fA-F]{3})$/', $value ) || preg_match( '/^#([0-9a-fA-F]{6})$/', $value ) );
+		return $this->validator ?? function ( $value ): bool {
+			if ( ! is_string( $value ) ) {
+				// translators: 1) is the name of the parameter.
+				$exception = new InvalidRestArgumentException( sprintf( __( 'Parameter `{%1$s}` must be a string.', 'the-events-calendar' ), $this->get_name() ) );
+				$exception->set_argument( $this->get_name() );
+				$exception->set_internal_error_code( 'tec_rest_invalid_hex_color_parameter' );
+
+				// translators: 1) is the name of the parameter.
+				$exception->set_details( sprintf( __( 'The parameter `{%1$s}` is not a string.', 'the-events-calendar' ), $this->get_name() ) );
+				throw $exception;
+			}
+
+			if ( ! ( preg_match( '/^#([0-9a-fA-F]{3})$/', $value ) || preg_match( '/^#([0-9a-fA-F]{6})$/', $value ) ) ) {
```

Comment:
```suggestion
			if ( ! ( preg_match( '/^#[0-9a-fA-F]{3}$|^#[0-9a-fA-F]{6}$/', $value ) ) ) {
```
can do it in one line 😉 
Treats single pipe (`|`) as "OR"